pkg/config/auth.go

```diff
@@ -0,0 +1,1156 @@
+	}
+	if remoteConfig.ExternalPhoneEnabled != nil {
+		result.Sms.EnableSignup = *remoteConfig.ExternalPhoneEnabled
+	}
+	if remoteConfig.ExternalAnonymousUsersEnabled != nil {
+		result.EnableAnonymousSignIns = *remoteConfig.ExternalAnonymousUsersEnabled
+	}
+	if remoteConfig.SmtpMaxFrequency != nil {
+		result.Email.MaxFrequency = time.Duration(*remoteConfig.SmtpMaxFrequency) * time.Second
+	}
+	if remoteConfig.MailerOtpLength != nil {
+		result.Email.OtpLength = cast.IntToUint(*remoteConfig.MailerOtpLength)
+	}
+	result.Email.OtpExpiry = cast.IntToUint(remoteConfig.MailerOtpExp)
+	if remoteConfig.MfaMaxEnrolledFactors != nil {
+		result.MFA.MaxEnrolledFactors = cast.IntToUint(*remoteConfig.MfaMaxEnrolledFactors)
+	}
+	if remoteConfig.MfaTotpEnrollEnabled != nil {
+		result.MFA.TOTP.EnrollEnabled = *remoteConfig.MfaTotpEnrollEnabled
+	}
+	if remoteConfig.MfaTotpVerifyEnabled != nil {
+		result.MFA.TOTP.VerifyEnabled = *remoteConfig.MfaTotpVerifyEnabled
+	}
+	if remoteConfig.MfaPhoneEnrollEnabled != nil {
+		result.MFA.Phone.EnrollEnabled = *remoteConfig.MfaPhoneEnrollEnabled
+	}
+	if remoteConfig.MfaPhoneVerifyEnabled != nil {
+		result.MFA.Phone.VerifyEnabled = *remoteConfig.MfaPhoneVerifyEnabled
+	}
+	if remoteConfig.MfaPhoneTemplate != nil {
+		result.MFA.Phone.Template = *remoteConfig.MfaPhoneTemplate
+	}
+	if remoteConfig.MfaPhoneMaxFrequency != nil {
+		result.MFA.Phone.MaxFrequency = time.Duration(*remoteConfig.MfaPhoneMaxFrequency) * time.Second
+	}
+	result.MFA.Phone.OtpLength = cast.IntToUint(remoteConfig.MfaPhoneOtpLength)
+	if remoteConfig.SmtpAdminEmail != nil {
+		result.Email.Smtp.AdminEmail = *remoteConfig.SmtpAdminEmail
+	}
+	if remoteConfig.SmtpHost != nil {
+		result.Email.Smtp.Host = *remoteConfig.SmtpHost
+	}
+	if remoteConfig.SmtpPass != nil {
+		result.Email.Smtp.Pass = *remoteConfig.SmtpPass
+	}
+	if remoteConfig.SmtpPort != nil {
+		if port, err := strconv.ParseUint(*remoteConfig.SmtpPort, 10, 16); err == nil {
+			result.Email.Smtp.Port = uint16(port)
+		}
+	}
+	if remoteConfig.SmtpUser != nil {
+		result.Email.Smtp.User = *remoteConfig.SmtpUser
+	}
+	if remoteConfig.SmtpSenderName != nil {
+		result.Email.Smtp.SenderName = *remoteConfig.SmtpSenderName
+	}
+	// Handle external providers
+	result.mapRemoteExternalProviders(remoteConfig)
+	// Handle email templates
+	result.mapRemoteEmailTemplates(remoteConfig)
+	// Handle hooks
+	result.mapRemoteHooks(remoteConfig)
+	// Handle SMS providers
+	result.mapRemoteSmsProviders(remoteConfig)
+
+	return result
+}
+
+func (a *auth) mapRemoteEmailTemplates(remoteConfig v1API.AuthConfigResponse) {
+	for name, template := range a.Email.Template {
+		switch name {
+		case "invite":
+			if remoteConfig.MailerSubjectsInvite != nil {
+				template.Subject = *remoteConfig.MailerSubjectsInvite
+			}
+			if remoteConfig.MailerTemplatesInviteContent != nil {
+				template.ContentPath = *remoteConfig.MailerTemplatesInviteContent
```

Comment:
We need to implement a new solution for this. The api returns file content but here we are assigning to a file path.

Perhaps writing to a predefined path would be sufficient.

Reply:
I'm also thinking to deprecate this `ContentPath` for simply `Content` field that accepts multi line string as input.

One inconsistency with the current `ContentPath` field is that it's relative to workdir and not the supabase dir like all other paths in config.toml.

Reply:
So the "template" would be directly embedded within the `config.toml`, I guess that would match what we have for the `mfa.phone` template message  🤔 

I do like to split my things up so having the templates in dedicated files in dedicated path if the content is more than a few lines long, but I can see the benefits of "having it all in a single place" as well. So this one is really a matter of preference.

If we wanna go that way basically, when we load the config, we look the content in the config, if not found, fallback on the path, and use this content sounds good to me.